### PR TITLE
Airgap

### PR DESCRIPTION
**Pull request by kiransynopsys**

### CODE REVIEW

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 1]
```diff
@@ -73,6 +73,35 @@ export class SynopsysBridge {
     }
 
     try {
+      if (inputs.ENABLE_NETWORK_AIR_GAP) {
+        if (inputs.SYNOPSYS_BRIDGE_PATH.length !== 0) {
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:77]: why are we validating like inputs.SYNOPSYS_BRIDGE_PATH.length !== 0, can't we check like if(inputs.SYNOPSYS_BRIDGE_PATH.length) 
is there any specific reason to check empty string with length

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 2]
```diff
@@ -73,6 +73,35 @@ export class SynopsysBridge {
...
+            );
+          }
+        } else {
+          executable = await this.setBridgeExecutablePath(
+            osName,
+            this.getBridgeDefaultPath()
+          );
+          if (!taskLib.exist(executable)) {
+            throw new Error(
+              "bridge_default_Path ".concat(
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:97]: we don't have any input variable like bridge_default_Path, can you make it Synopsys Bridge default path does not exist

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 3]
```diff
@@ -73,6 +73,35 @@ export class SynopsysBridge {
     }
 
     try {
+      if (inputs.ENABLE_NETWORK_AIR_GAP) {
+        if (inputs.SYNOPSYS_BRIDGE_PATH.length !== 0) {
+          executable = await this.setBridgeExecutablePath(
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:78]: can we validate SYNOPSYS_BRIDGE_PATH as well, and log the error 
and if executable not found, throw error relatable to that

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 4]
```diff
@@ -73,6 +73,35 @@ export class SynopsysBridge {
...
+            osName,
+            inputs.SYNOPSYS_BRIDGE_PATH
+          );
+          if (!taskLib.exist(executable)) {
+            throw new Error(
+              "synopsys_bridge_path ".concat(
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:84]: you can pick this value from constants

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 5]
```diff
@@ -73,6 +73,35 @@ export class SynopsysBridge {
...
+                inputs.SYNOPSYS_BRIDGE_PATH,
+                " does not exists"
+              )
+            );
+          }
+        } else {
+          executable = await this.setBridgeExecutablePath(
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:91]: can we validate default bridge path and log the error if it doesn't exist
and if executable not found, throw error relatable to that

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 6]
```diff
@@ -301,11 +329,16 @@ export class SynopsysBridge {
         );
       }
     } else {
-      console.info(
+      taskLib.debug(
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:332]: why you're replacing console.info with taskLib.debug here, we are showing this message in pipeline

File: synopsys-task/task.json [hunk 7]
```diff
@@ -246,6 +246,33 @@
...
+      "name": "network_air_gap",
+      "type": "string",
+      "label": "Network Air Gap",
+      "defaultValue": "",
+      "required": false,
+      "helpMarkDown": "enables airgap's configuration",
+      "groupName": "blackduck"
+    },
+    {
+      "name": "network_air_gap",
```
Comment (spurohitsynopsys) [inline synopsys-task/task.json:269]: why adding network_air_gap twice ?

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 8]
```diff
@@ -422,6 +435,44 @@ export class SynopsysBridge {
...
+        const htmlResponse = (await httpResponse.readBody()).trim();
+        const lines = htmlResponse.split("\n");
+        for (const line of lines) {
+          if (line.includes("Synopsys Bridge Package")) {
+            const newerVersion = line.split(":")[1].trim();
+            return newerVersion;
+          }
+        }
+      } else {
+        taskLib.error("Unable to locate version url");
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts]: can we make more relatable error message like: Unable to fetch the latest version from artifactory
we can check once with team.

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 9]
```diff
@@ -469,4 +520,18 @@ export class SynopsysBridge {
...
     return bridgeDownloadUrl;
   }
+
+  async setBridgeExecutablePath(
+    osName: string,
+    filePath: string
+  ): Promise<string> {
+    if (osName === "win32") {
+      this.bridgeExecutablePath = filePath
+        .concat("\\synopsys-bridge")
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:558]: you can use path.join as used in executeBridgeCommand method

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 10]
```diff
@@ -469,4 +520,18 @@ export class SynopsysBridge {
     }
     return bridgeDownloadUrl;
   }
+
+  async setBridgeExecutablePath(
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:552]: we are setting executable path when airgap is false, you can use this method there as well in executeBridgeCommand method

**Review by kiransynopsys — COMMENTED** [review]
updated code.

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 11]
```diff
@@ -469,4 +522,45 @@ export class SynopsysBridge {
...
+    osName: string,
+    filePath: string
+  ): Promise<string> {
+    if (osName === "win32") {
+      this.bridgeExecutablePath = path.join(
+        filePath,
+        constants.SYNOPSYS_BRIDGE_EXECUTABLE_WINDOWS
+      );
+    } else {
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:558]: use else if and check for linux and mac OS only, since we are not supporting any other OS

Comment (kiransynopsys) [reply synopsys-task/src/synopsys-task/synopsys-bridge.ts:558]: we can use the code, no need to change.

**Review by spurohitsynopsys — APPROVED** [review]

**Review by lsynopsys — APPROVED** [review]

**Pull request merged**